patchwork-events-world/src/main/java/com/patchworkmc/mixin/event/world/MixinMinecraftServer.java

```diff
@@ -0,0 +1,112 @@
+import java.util.Map;
+
+import net.minecraftforge.common.MinecraftForge;
+import net.minecraftforge.event.world.WorldEvent;
+import org.spongepowered.asm.mixin.Final;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Redirect;
+
+import net.minecraft.server.MinecraftServer;
+import net.minecraft.server.ServerTask;
+import net.minecraft.server.world.ServerWorld;
+import net.minecraft.util.NonBlockingThreadExecutor;
+import net.minecraft.world.dimension.DimensionType;
+
+@Mixin(MinecraftServer.class)
+public abstract class MixinMinecraftServer extends NonBlockingThreadExecutor<ServerTask> {
+	public MixinMinecraftServer(String name) {
+		super(name);
+	}
+
+	@Shadow
+	@Final
+	private Map<DimensionType, ServerWorld> worlds;
+
+	// Should get called once per loop, regardless of which if branch it takes.
+	//	@Inject(
+	//		method = "createWorlds",
+	//		slice = @Slice(
+	//			from = @At(value = "INVOKE", target = "java/util/Iterator.hasNext ()Z")
+	//		),
+	//		at = @At(value = "JUMP", opcode = Opcodes.GOTO),
+	//		locals = LocalCapture.CAPTURE_FAILHARD
+	//	)
+	//	private void hookCreateWorlds(WorldSaveHandler worldSaveHandler, LevelProperties properties, LevelInfo levelInfo, WorldGenerationProgressListener worldGenerationProgressListener, CallbackInfo ci, ServerWorld serverWorld, ServerWorld serverWorld2, Iterator var7, DimensionType dimensionType) {
+	//		MinecraftForge.EVENT_BUS.post(new WorldEvent.Load(this.worlds.get(dimensionType)));
+	//	}
+
+	// Alternatively, mixin to the put call, and special case overworld.
+	// Perhaps move the special case outside of the loop?
+	@Redirect(method = "createWorlds", at = @At(value = "INVOKE", target = "java/util/Iterator.next ()Ljava/lang/Object;"))
+	private Object proxyNextWorldToSpecialCaseOverworld(Iterator iterator) {
+		DimensionType type = (DimensionType) iterator.next();
+
+		if (type == DimensionType.OVERWORLD) {
+			MinecraftForge.EVENT_BUS.post(new WorldEvent.Load(this.worlds.get(type)));
+		}
+
+		return type;
+	}
+
+	@Redirect(method = "createWorlds", at = @At(value = "INVOKE", target = "java/util/Map.put (Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;", ordinal = 1))
+	private Object proxyPutWorld(Map worlds, Object type, Object world) {
+		worlds.put(type, world);
+		MinecraftForge.EVENT_BUS.post(new WorldEvent.Load((ServerWorld) world));
+		return world;
+	}
+
+	// TODO: Should we Inject into ServerWorld.close instead? Currently, this follows Forge's patch location.
+	@Redirect(method = "shutdown", at = @At(value = "INVOKE", target = "net/minecraft/server/world/ServerWorld.close ()V"))
+	private void proxyClose(ServerWorld world) throws IOException {
+		MinecraftForge.EVENT_BUS.post(new WorldEvent.Unload(world));
+		world.close();
+	}
+
+	@Shadow
+	private int ticks;
+
+	// TODO: DimensionManager, and move this into a seperate module
```

Comment:
I ended up creating some mixins to handle some DimensionManager patches. I can totally remove them from this PR, just figured I'd leave them in in case they'd be useful in the future.